### PR TITLE
Add VC Presentation for Native Wallet docs.

### PR DESCRIPTION
**Pull request by BigBlueHat**
@brianorwhatever and I are working toward launching this before Wednesday of next week.

We also started some Mermaid.js (#30) for this section:
```mermaid
sequenceDiagram
  # User visits Relying Party website and clicks something. Then this happens...

  participant site as OpenBadge Verifier Site
  participant app as Native Wallet App
  participant chapi as User's Browser / CHAPI
  participant exchanger as VC-API Exchanger

  site ->> exchanger: 1. creates an exchange within a workflow
  exchanger ->> site: 2. exchange URL/ID returned (in some JSON)
  site ->> chapi: 3. triggers CHAPI `get()` request w/blank VP + `protocols.vcapi` containing URL
  chapi ->> app: 4. triggers app URL to open app w/`?request=`
  app ->> exchanger: 5. uses VC-API  URL from `?request=` to send VPR
  exchanger ->> app: 6. returns VPR to wallet
  app ->> exchanger: 7. responds with VP
  exchanger ->> app: 9. sends success response to wallet
  # Webhooks here maybe?
  loop poll exchange status
    site ->> exchanger: exchange complete?
    exchanger ->> site: ...wait for it...
  end
```

### CODE REVIEW

**Comment by dlongley** [comment]
Still reviewing this, but it struck me that (maybe?) the diagram would look better with the position of `Native Wallet App` and `User's Browser / CHAPI` positionally reversed.

**Comment by dlongley** [comment]
The `VP` doesn't have to be blank (and it's a blank `VPR` really, it just is the value of the `VerifiablePresentation` property for arcane reasons). So maybe we leave that bit out -- it's not relevant to native wallets anyway.

**Comment by dlongley** [comment]
> 5. uses VC-API URL from `?request=` to send VPR

I think we're calling this URL too many different things. This is the "exchange URL" from the beginning; it will be found under `protocols.vcapi` ... but maybe that's not relevant?

Also, the VPR isn't being sent an empty `{}` JSON is POSTed to the exchange URL. Then, as step 6 says, the VPR is returned as a response.

File: developers/wallets/native/index.md [hunk 1]
```diff
@@ -219,3 +219,105 @@ object containing the following details:
...
+Depending on the protocols supported, the value of request will contain one
+or more encoded JSON objects with nifty details...
+
+...explain when this happens...maybe with mermaid chart?
+
+### VC-API
+
+The `vcapi` property within the object parsed above will contain a URL the wallet can use to handle that request.
+
+To respond with one or more credentials via a VC-API exchanger, the wallet MUST first send a `POST` request
```
Comment (dlongley) [inline developers/wallets/native/index.md:244]: And sometimes no credentials at all! (DID Authn only).

File: developers/wallets/native/index.md [hunk 2]
```diff
@@ -219,3 +219,105 @@ object containing the following details:
...
+to the URL (extracted from `protocols.vcapi` above) with an empty JSON object
+(`{}`) in the body of the request to attempt to download the VC to be stored.
```
Comment (dlongley) [inline developers/wallets/native/index.md:246]:
```suggestion
(`{}`) in the body of the request to attempt to get a verifiable presentation request with
the details about what the verifier wants.
```

File: developers/wallets/native/index.md [hunk 3]
```diff
@@ -219,3 +219,105 @@ object containing the following details:
...
+            ]
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+The wallet should fullfil the request by sending another `POST` request to the
```
Comment (dlongley) [inline developers/wallets/native/index.md:287]:
```suggestion
The wallet should fulfil the request by sending another `POST` request to the
```

**Review by dlongley — CHANGES_REQUESTED** [review]

**Comment by BigBlueHat** [comment]
> Still reviewing this, but it struck me that (maybe?) the diagram would look better with the position of `Native Wallet App` and `User's Browser / CHAPI` positionally reversed.

This look better?
image

**Comment by BigBlueHat** [comment]
> The `VP` doesn't have to be blank (and it's a blank `VPR` really, it just is the value of the `VerifiablePresentation` property for arcane reasons). So maybe we leave that bit out -- it's not relevant to native wallets anyway.

Are you referring to the sending of the empty JSON object? Meaning, it'd be valid to send a completely empty POST to the exchanges endpoint?

File: developers/wallets/native/index.md (hunk 3 above)
Comment (BigBlueHat) [reply developers/wallets/native/index.md:287]: Almost...`fulfill` 😀

File: developers/wallets/native/index.md (hunk 1 above)
Comment (BigBlueHat) [reply developers/wallets/native/index.md:244]: @dlongley How about "zero or more"?

**Comment by BigBlueHat** [comment]
> > The `VP` doesn't have to be blank (and it's a blank `VPR` really, it just is the value of the `VerifiablePresentation` property for arcane reasons). So maybe we leave that bit out -- it's not relevant to native wallets anyway.
> 
> Are you referring to the sending of the empty JSON object? Meaning, it'd be valid to send a completely empty POST to the exchanges endpoint?

Never mind. I realized you're referring to the Mermaid chart... I'll get that in as it's own commit, so it can be reviewed directly.

File: developers/wallets/native/index.md (hunk 1 above)
Comment (dlongley) [reply developers/wallets/native/index.md:244]: "zero or more" works for me.

**Comment by BigBlueHat** [comment]
@dlongley I made a few more changes to the chart and also improved the viewing experience for mermaid charts across the site. Here's a demo: http://localhost:8080/developers/wallets/native/#verifiable-credential-presentation

**Review by dlongley — APPROVED** [review]
That link is to localhost ... which either means you want me to clone and run to see or the demo is elsewhere and the link needs updating. Either way, I'm inclined to approve this PR to get this material in. Thanks!

**Comment by dlongley** [comment]
JFYI, the mermaid diagram looks like this for me (dark mode):

mermaid-diagram

It's not top priority to fix this, but would be nice if we could somehow.

**Comment by BigBlueHat** [comment]
@dlongley good catch. It's definitely non-trivial to fix this...sadly. Mermaid has themes (and one that's dark mode friendly), but it's gonna take a bit to do that dynamically, etc. I'll file and issue and try to get to it soon.

File: developers/wallets/native/index.md [hunk 4]
```diff
@@ -219,3 +219,129 @@ object containing the following details:
...
+  # User visits a Verifier's website and interacts with it to trigger a verification
+
+  participant site as Verifier Site
+  participant chapi as User's Browser / CHAPI
+  participant app as Native Wallet App
+  participant exchanger as VC-API Exchanger
+
+  site ->> exchanger: 1. creates an exchange within a workflow
+  exchanger ->> site: 2. JSON response contains the exchange URL/ID
+  site ->> chapi: 3. triggers CHAPI `get()` request w/blank VP + `protocols.vcapi` containing URL
```
Comment (brianorwhatever) [inline developers/wallets/native/index.md:250]:
```suggestion
  site ->> chapi: 3. triggers CHAPI `get()` request w/blank `VerifiablePresentation` object + `protocols.vcapi` containing URL
```

**Review by brianorwhatever — CHANGES_REQUESTED** [review]
This looks good other than the first paragraphs under "Verifiable Credential Presentation"
I think there is just a little cleanup needed there..

I also have a note for number 3. in the mermaid chart. The "VP" in this case is actually a blank `VerifiablePresentation` object which is actually slightly different than a true VP.. It's CHAPI communicating a ______ of type `VerifiablePresentation` is happening.. Still shaky on the exact meaning. @dlongley would have to fill in the blanks here.

**Comment by brianorwhatever** [comment]
Let's address this small change request and get this merged in @BigBlueHat

File: developers/wallets/native/index.md (hunk 4 above)
Comment (BigBlueHat) [reply developers/wallets/native/index.md:250]: @brianorwhatever we're going to need code examples for this also, 'cause it's hard to even guess at what a "blank `VerifiablePresentation` object might be...but I'll commit the text change.

**Comment by BigBlueHat** [comment]
> Let's address this small change request and get this merged in @BigBlueHat

Cleaned that unfinished stuff up, and made a couple other smaller improvements. I think it's ready.

**Comment by BigBlueHat** [comment]
@brianorwhatever can you re-review or change the "requested changes" status at least? I think this is ready and would like to move on to fixing it in further PRs if/when we find more to change. Thanks!

**Review by brianorwhatever — APPROVED** [review]

**Pull request merged**